client/src/app/app.component.ts

```diff
@@ -13,24 +16,34 @@ interface User {
   templateUrl: './app.component.html',
   styleUrls: ['./app.component.scss'],
 })
-export class AppComponent implements OnInit {
+export class AppComponent implements OnInit, OnDestroy {
   users: User[] = [];
-  constructor(private apollo: Apollo, public translate: TranslateService) {}
+  subscriptions: Subscription[] = [];
+  logged = false;
+  constructor(private apollo: Apollo, public translate: TranslateService, private authService: AuthService) { }
 
   ngOnInit() {
-    this.apollo
-      .watchQuery({
-        query: gql`
-          {
-            users {
-              username
-            }
-          }
-        `,
-      })
-      .valueChanges.subscribe((result) => {
-        // @ts-ignore
-        this.users = result.data && result.data.users;
+
+    const isAuthenticatedSubscription = this.authService.isAuthenticated()
+      .pipe(distinctUntilChanged())
+      .subscribe(isAuthenticated => {
+        this.logged = isAuthenticated;
       });
+
+    this.subscriptions = [...this.subscriptions, isAuthenticatedSubscription];
+
+    this.authService.autoLogin()
+  }
+
+  logout() {
+    this.authService.logout();
+  }
+
+  ngOnDestroy(): void {
```

Comment:
I suggest to create abstract class that you can implement in any component and manage subscriptions with takeUntil(this.$destroy) instead of unsubbing manually in every component
`export abstract class SubscriptionDestroyer implements OnDestroy {
  protected destroy$ = new Subject<void>();

  public ngOnDestroy(): void {
    this.destroy$.next();
  }
}`